### PR TITLE
test: replace render with po in NnsProposals spec

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

In the near future the actionable proposals will be selected by default. To be able to switch between all and actionable proposals in the spec file, they should use page object for rendering.

# Changes

- Add missing PO getters.
- Switch from `render()` to using a page object.

# Tests

- Yes.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts [hunk 1]
```diff
@@ -27,11 +26,7 @@ import { JestPageObjectElement } from "$tests/page-objects/jest.page-object";
 import { NnsProposalListPo } from "$tests/page-objects/NnsProposalList.page-object";
 import { runResolvedPromises } from "$tests/utils/timers.test-utils";
 import type { HttpAgent } from "@dfinity/agent";
-import {
-  GovernanceCanister,
-  type Proposal,
-  type ProposalInfo,
-} from "@dfinity/nns";
+import { GovernanceCanister, type ProposalInfo } from "@dfinity/nns";
 import { render, waitFor } from "@testing-library/svelte";
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/NnsProposals.spec.ts:30]: FYI, if you import { render } from "[$tests/utils/svelte.test-utils](https://github.com/dfinity/nns-dapp/blob/main/frontend/src/tests/utils/svelte.test-utils.ts)" instead, then you won't have issues if you call render multiple times in the same test.

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts [hunk 2]
```diff
@@ -109,34 +105,44 @@ describe("NnsProposals", () => {
...
 
-      it("should render proposals", () => {
-        const { getByText } = render(NnsProposals);
-
+      it("should render proposals", async () => {
+        const po = await renderComponent();
         const firstProposal = mockProposals[0] as ProposalInfo;
         const secondProposal = mockProposals[1] as ProposalInfo;
+
+        expect(await po.getProposalCardPos()).toHaveLength(2);
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/NnsProposals.spec.ts:139]: Perhaps assign the result of `await po.getProposalCardPos()` to a variable so you don't have to repeat it again?

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts [hunk 3]
```diff
@@ -264,32 +265,25 @@ describe("NnsProposals", () => {
...
+      it("should render proposals", async () => {
         mockLoadProposals();
 
-        const { getByText } = render(NnsProposals);
-
+        const po = await renderComponent();
         const firstProposal = mockProposals[0] as ProposalInfo;
         const secondProposal = mockProposals[1] as ProposalInfo;
+
+        expect(await po.getProposalCardPos()).toHaveLength(2);
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/NnsProposals.spec.ts:275]: Perhaps assign the result of `await po.getProposalCardPos()` to a variable so you don't have to repeat it again?

File: frontend/src/tests/page-objects/NnsProposalFilters.page-object.ts [hunk 4]
```diff
@@ -11,12 +11,20 @@ export class NnsProposalFiltersPo extends BasePageObject {
     return new NnsProposalFiltersPo(element.byTestId(NnsProposalFiltersPo.TID));
   }
 
+  getFilterByTopicsButtonPo(): PageObjectElement {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/NnsProposalFilters.page-object.ts:14]: The name of the method suggests this returns type `ButtonPo` but it doesn't.
Also, typically `PageObjectElement` objects should stay internal to page objects and not be returned.
Could you just `return this.getButton("filters-by-topics")` here?

File: frontend/src/tests/page-objects/NnsProposalFilters.page-object.ts [hunk 5]
```diff
@@ -11,12 +11,20 @@ export class NnsProposalFiltersPo extends BasePageObject {
...
+    return this.root.byTestId("filters-by-topics");
+  }
+
+  getFilterByStatusButtonPo(): PageObjectElement {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/NnsProposalFilters.page-object.ts:18]: same

File: frontend/src/tests/page-objects/NnsProposalList.page-object.ts [hunk 6]
```diff
@@ -16,6 +17,10 @@ export class NnsProposalListPo extends BasePageObject {
     return SkeletonCardPo.under(this.root);
   }
 
+  getListLoaderSpinnerPo(): PageObjectElement {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/NnsProposalList.page-object.ts:20]: This also doesn't return a PO.
How about:
```
hasListLoaderSpinner(): Promise<boolean> {
  return this.isPresent("next-page-sns-proposals-spinner");
}
```

File: frontend/src/tests/lib/pages/NnsProposals.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/NnsProposals.spec.ts]: Good point!

**Comment by mstrasinskis** [comment]
The pr has been updated. @dskloetd, could you place take another look?

**Review by dskloetd — APPROVED** [review]
Thanks!

**Pull request merged**